common/lib/xmodule/xmodule/video_module/video_handlers.py

```diff
@@ -199,6 +200,38 @@ def get_static_transcript(self, request, transcripts):
                 )
         return response
 
+    def get_val_transcript(self, language_code):
+        """
+        Gets video transcript from edx-val if found, otherwise returns an HTTP 404 Response.
+
+        Arguments:
+            language_code(unicode): Language code of the requested transcript.
+
+        Returns:
+            An HTTP Response containing the transcript content or an HTTP 404 Response.
+        """
+        response = Response(status=404)
+        transcript = get_video_transcript_content(
+            course_id=self.course_id,
+            language_code=language_code,
+            edx_video_id=self.edx_video_id,
+            youtube_id_1_0=self.youtube_id_1_0,
+            html5_sources=self.html5_sources,
+        )
+        if transcript:
+            # Setting `self.transcript_language` is necessary here otherwise /download dispatch
+            # would never know of the current transcript language.
+            if language_code != self.transcript_language:
+                self.transcript_language = language_code
```

Comment:
Feeling little bit concerned about this change.

Reply:
Fields are going to be save after the invocation of this handler: https://github.com/edx/XBlock/blob/master/xblock/runtime.py#L1039